src/core/viewports/orbit-viewport.js

```diff
@@ -43,30 +45,35 @@ export default class OrbitViewport extends Viewport {
     translationY = 0, // in pixels
     zoom = 1
   }) {
-    const eye = vec3_add([], lookAt, [0, 0, distance]);
-    vec3_rotateX(eye, eye, lookAt, rotationX / 180 * Math.PI);
-    vec3_rotateY(eye, eye, lookAt, rotationY / 180 * Math.PI);
+    const rotationMatrix = mat4_rotateX([], createMat4(), -pitchAngle / 180 * Math.PI);
+    if (orbitAxis === 'Z') {
+      mat4_rotateZ(rotationMatrix, rotationMatrix, -orbitAngle / 180 * Math.PI);
+    } else {
+      mat4_rotateY(rotationMatrix, rotationMatrix, -orbitAngle / 180 * Math.PI);
+    }
+
+    const transformMatrix = mat4_translate([], createMat4(),
+      [translationX / width * 2, translationY / height * 2, 0]);
+    mat4_scale(transformMatrix, transformMatrix, [zoom, zoom, zoom]);
+    mat4_multiply(rotationMatrix, rotationMatrix, transformMatrix);
```

Comment:
This is incorrect. `translation` and `scale` were applied to screen space because when panning or scrolling to zoom, the "grabbed" pixel must remain under the cursor. With this change, depending on the orbit angle, the model may travel in the opposite direction of the drag.

I understand the problem of transforms in screen space, and i agree that we need to change it. The old implementation was to avoid tackling the difficult math problem of translating 2d interaction back to 3d camera movement. Now that we're at it, might as well do it right.

Reply:
yeah, you are right, the code is still not good enough. I will fix it in later PR instead of a big one with everything inside. This problem is more difficult, so need move forward gradually, but at least we have a better direction to go.